src/modules/sensors/sensors.cpp

```diff
@@ -1714,9 +1762,51 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 			}
 
 			_last_adc = t;
+		}
+	}
+}
 
+void
+Sensors::gps_poll(struct vehicle_gps_position_s gps[2])
```

Comment:
Rather than doing this here the two topics should be advertised and both consumed by the estimator, which has the required information to decide which one he would prefer to use.